src/btree/bt_prefetch.c

```diff
@@ -95,11 +96,20 @@ __wt_prefetch_page_in(WT_SESSION_IMPL *session, WT_PREFETCH_QUEUE_ENTRY *pe)
 
     WT_STAT_CONN_INCR(session, block_prefetch_pages_read);
 
+    /*
+     * If pre-fetching for a checkpoint handle, ensure the page-in checkpoint reader visibility
+     * check isn't worried about our visibility setup.
+     */
+    if (WT_READING_CHECKPOINT(session))
+        F_SET(session->txn, WT_TXN_IS_CHECKPOINT);
```

Comment:
Setting the prefetch session transaction state to `WT_TXN_IS_CHECKPOINT` will lead to accessing the values of the transaction structure that are not populated. For example, in the function `__txn_visible_all_id` the structure members are accessed.

I prefer the solution to avoid performing any visibility checks if it is a prefetch session to fix these problems.

Reply:
That makes sense, I'll try to come up with an alternative fix and test that first.